File: groom/GameJam.js
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

let input = [];
rl.on("line", (line) => {
    input.push(line);
}).on("close", () => {
    const n = parseInt(input[0]);
    const coords = input[1].split(" ").map((str) => parseInt(str));
    const matrix = input.slice(2, 2 + n).map((line) => line.split(" "));

    function getScore(x, y, cnt, dir, visited) {
        if (visited[x][y]) {
            return 0;
        }
        visited[x][y] = true;

        if (cnt === 0) {
            const parts = matrix[x][y].split("");
            cnt = parseInt(parts.slice(0, parts.length - 1).join(""));
            dir = parts[parts.length - 1];
        }

        if (dir === "U") {
            return (
                1 +
                getScore(x - 1 >= 0 ? x - 1 : n - 1, y, cnt - 1, dir, visited)
            );
        } else if (dir === "D") {
            return 1 + getScore((x + 1) % n, y, cnt - 1, dir, visited);
        } else if (dir === "L") {
            return (
                1 +
                getScore(x, y - 1 >= 0 ? y - 1 : n - 1, cnt - 1, dir, visited)
            );
        } else if (dir === "R") {
            return 1 + getScore(x, (y + 1) % n, cnt - 1, dir, visited);
        }
    }

    const score = [];

    for (let i = 0; i < coords.length; i++) {
        coords[i]--;
    }

    for (let i = 0; i < coords.length; i += 2) {
        const x = coords[i];
        const y = coords[i + 1];
        const visited = new Array(n)
            .fill(false)
            .map(() => new Array(n).fill(false));
        score.push(getScore(x, y, 0, "", visited));
    }

    if (score[0] < score[1]) {
        console.log("player", score[1]);
    } else {
        console.log("goorm", score[0]);
    }
});

File: groom/GameJam.py
import sys

input = sys.stdin.readline


def solve():
    n = int(input())
    xy_coords = [tuple(map(int, input().split())) for _ in range(2)]
    grid = [list(input().split()) for _ in range(n)]

    scores = []

    def calculate_score(x, y, count, direction):
        if visited[x][y]:
            return 0
        visited[x][y] = True
        if count == 0:
            count, direction = int(grid[x][y][:-1]), grid[x][y][-1]
        if direction == "U":
            return 1 + calculate_score(
                x - 1 if x - 1 >= 0 else n - 1, y, count - 1, direction
            )
        elif direction == "D":
            return 1 + calculate_score((x + 1) % n, y, count - 1, direction)
        elif direction == "L":
            return 1 + calculate_score(
                x, y - 1 if y - 1 >= 0 else n - 1, count - 1, direction
            )
        elif direction == "R":
            return 1 + calculate_score(x, (y + 1) % n, count - 1, direction)

    for x, y in xy_coords:
        x -= 1
        y -= 1
        visited = [[False] * n for _ in range(n)]
        scores.append(calculate_score(x, y, 0, ""))

    if scores[0] < scores[1]:
        print("player", scores[1])
    else:
        print("goorm", scores[0])


solve()
